app/src/main/java/MainMenu.java

```diff
@@ -0,0 +1,121 @@
+import static com.raylib.Jaylib.*;
+
+import com.raylib.Jaylib;
+
+/** The MainMenu class manages the game's main menu options using the Singleton Pattern. */
+public class MainMenu {
+
+    // Instance variable for the Singleton Pattern
+    private static final MainMenu instance = new MainMenu();
+
+    private Util.Background background;
+
+    UtilButton singleplayerButton;
+    UtilButton multiplayerButton;
+    UtilButton exitButton;
+
+    // Private constructor for the Singleton Pattern
+    private MainMenu() {
+
+        // Create Buttons
+        singleplayerButton =
+                new UtilButton(
+                        (float) Constants.WIDTH / 2 - ((float) 250 / 2),
+                        (float) Constants.HEIGHT / 2 - ((float) 50 / 2),
+                        250,
+                        50,
+                        "SINGLEPLAYER");
+        multiplayerButton =
+                new UtilButton(
+                        singleplayerButton.x(),
+                        singleplayerButton.y() + singleplayerButton.height(),
+                        250,
+                        50,
+                        "MULTIPLAYER");
+        exitButton =
+                new UtilButton(
+                        multiplayerButton.x(),
+                        multiplayerButton.y() + multiplayerButton.height(),
+                        250,
+                        50,
+                        "EXIT");
+
+        createBackground();
+    }
+
+    /**
+     * Returns the sole instance of the MainMenu class.
+     *
+     * @return Instance of MainMenu
+     */
+    public static MainMenu getInstance() {
+        return instance;
+    }
+
+    /** Displays the background window with menu options. */
+    public void showBackground() {
+        DrawTexture(background.texture, 0, 0, WHITE);
+
+        singleplayerButton.draw();
+        multiplayerButton.draw();
+        exitButton.draw();
+    }
+
+    public void update() {
+        checkInput();
+    }
```

Comment:
Hier fehlt ein JavaDoc außerdem sollte der Code aus checkInput hier rein oder in game direkt checkInput aufrufen, eine Methode die nur dafür da ist eine andere Methode aufzurufen ist irgendwie nicht wirklich effektiv.

Reply:
Ich weiss, was du meinst, aber jede Klasse ist für sich selber verantwortlich.

Was meine ich damit?
Nehmen wir alle Objekte bei der die Variablen aktualisiert werden.
Jede klasse sollte eine update Methode enthalten, die dann in der gameloop aufgerufen wird.

Sagen wir, wir haben eine weitere Methode bei der etwas abgefragt wird, dann würden wir diese auch in der Update Methode der Klasse MainMenu einfügen, damit alles übersichtlich ist, ansonsten hätte man, wenn man diese in der game (Update-Methode einfügen würde) zu viele unabhängige Methoden.


Aber mir ist es egal :)

Reply:
Ich verstehe den Gedanken dahinter wenn man mehrere Methoden hat die aufgerufen werden, aber stand jetzt haben wir ja nur eine Methode und da leidet dann die Übersichtlichkeit nicht wirklich drunter.

Reply:
ok :)